include/universal/utility/evaluate_closure_of_number_systems.hpp

```diff
@@ -0,0 +1,202 @@
+#pragma once
+
+// Copyright (C) 2017 Stillwater Supercomputing, Inc.
+// SPDX-License-Identifier: MIT
+//
+// Author: Colby Wirth
+// Version: 18 April 2025
+// 
+// About:
+// This utility calls the buildClosurePlot function from genereateClosurePlots.hpp 
+// and generates the closure statistics for a templatized number system. 
+//
+//
+// ****************************************************************
+// *** Use 'processASystem()' function call to use this utility ***
+// ****************************************************************
+// 
+// This file is part of the universal numbers project, which is released under an MIT Open Source license.
+
+#include <iostream>
+#include <string>
+#include <string_view>
+#include <fstream>
+
+#include <universal/number/posit/posit.hpp>
+#include <universal/number/cfloat/cfloat.hpp>
+#include <universal/number/lns/lns.hpp>
+#include <universal/utility/generateClosurePlots.hpp>
+
+
+// the data for each operation in a buildClosurePlots is stored in this struct
+struct operationResults{
+    float TotalOps;
+    std::string Exact;
+    std::string Approximate;
+    std::string Overflow;
+    std::string Underflow;
+    std::string Saturate;
+    std::string narOrNan;
+};
+
+// a storeResults struct stores the operationResult structs for a number system
+struct storeResults{
+    std::string sys_name ;
+    operationResults addition;
+    operationResults subtraction;
+    operationResults multiplication;
+    operationResults division;
+};
+
+///<summary>
+/// helper function for processASystem()
+/// formats and prints the results
+/// /</summary>
+///
+///<param> name="results" a struct that stores all of the statistics of the closure plot for a number system </param>
+/// <returns> void </returns>
+/// 
+void printResults(const storeResults& results) {
+    constexpr unsigned data_width = 15;
```

Comment:
Excellent, that is a good use of constexpr. 

I have gotten into the habit to using CAPITAL LETTERS for constants, so that you 'know' that we are dealing with a constant. But I am not consistent with it, so you will see constexpr names like bTrace.

for COL_WIDTH, and DATA_WIDTH, it makes sense to me. I leave it to your decision if you rename it or not.